metaflow/plugins/metadata/local.py

```diff
@@ -109,28 +112,62 @@ def register_metadata(self, run_id, step_name, task_id, metadata):
     def _get_object_internal(
         cls, obj_type, obj_order, sub_type, sub_order, filters, attempt, *args
     ):
+        # This is guaranteed by MetaflowProvider.get_object(), sole intended caller
+        if obj_type in ("metadata", "self"):
+            raise MetaflowInternalError(msg="Type %s is not allowed" % obj_type)
+
+        assert obj_type in ("root", "flow", "run", "step", "task", "artifact"), (
```

Comment:
Why not keep it a `MetaflowInternalError`? I think we have tried to avoid asserts in general as they don't really give a huge amount of information. Am fine either ways but for consistency, it seems preferable to return an exception than have an assert (yes yes, asserts raise exceptions).

Reply:
Don't feel strongly about it, updated!